docs/markdown/Reference-manual.md

```diff
@@ -737,6 +737,31 @@ arguments. The following keyword arguments are supported:
 
  To leave any of these three as the default, specify `false`.
 
+- `rename` if specified renames each source file into corresponding file
+  from `rename` list. Nested paths are allowed and they are joined with
+  `install_dir`. Length of `rename` list must be equal to the number of sources.
+  *(added 0.46.0)*
+
+#### Rename examples
+
+Install `file.txt` into `{datadir}/{projectname}/new-name.txt`
+```meson
+install_data('file.txt', rename : 'new-name.txt')
+```
+
+Install
+- `file1.txt` into `share/myapp/dir1/data.txt`
+- `file2.txt` into `share/myapp/dir2/data.txt`
+```meson
+install_data(['file1.txt', 'file2.txt'],
+             rename : ['dir1/data.txt', 'dir2/data.txt'],
+             install_dir : 'share/myapp')
+```
+
+Not allowed because of different sources and rename sizes:
+- `install_data('file.txt', rename : ['name1.txt', 'name2.txt'])`
+- `install_data(['file1.txt', 'file2.txt'], rename : 'name.txt')`
+
```

Comment:
This should be in Installing.md. All examples and usage instructions should be there.